eta/classifiers/tfslim_classifiers.py

```diff
@@ -250,50 +253,126 @@ def features_dim(self):
+            img: an image
 
         Returns:
             an `eta.core.data.AttributeContainer` instance containing the
                 predictions
         '''
-        if self.generates_features:
-            probs, fvec = self._evaluate(
-                img, [self._output_op, self._features_op])
-            self._last_features = fvec
+        return self._predict([img])[0]
+
+    def predict_all(self, imgs):
+        '''Performs prediction on the given tensor of images.
+
+        Args:
+            imgs: a list (or n x h x w x 3 tensor) of images
+
+        Returns:
+            a list of `eta.core.data.AttributeContainer` instances describing
+                the predictions for each image
+        '''
+        return self._predict(imgs)
+
+    def _predict(self, imgs):
+        # Perform preprocessing
+        imgs = self.preprocessing_fcn(imgs)
+
+        # Perform inference
+        if self.exposes_features:
+            features, probs = self._evaluate(
+                imgs, [self._features_op, self._output_op])
         else:
-            probs = self._evaluate(img, [self._output_op])[0]
+            features = None
+            probs = self._evaluate(imgs, [self._output_op])[0]
+
+        # Parse predictions
+        max_num_preds = 0
+        predictions = []
+        for probsi in probs:
+            # Filters predictions, if necessary
+            predsi, keepi = self._parse_prediction(probsi)
+            if keepi:
+                max_num_preds = 1
 
-        return self._parse_prediction(probs)
+            # Record predictions
+            predictions.append(predsi)
 
-    def _evaluate(self, img, ops):
-        imgs = self.preprocessing_fcn([img])
+        # Trim unnecessary dimensions
+        probs = probs[:, np.newaxis, :]
+        probs = probs[:, :max_num_preds, :]
+
+        # Save data, if necessary
+        if self.exposes_features:
+            self._last_features = features  # n x features_dim
+        self._last_probs = probs  # n x 1 x num_classes
+
+        return predictions
+
+    def _evaluate(self, imgs, ops):
         in_tensor = self._input_op.outputs[0]
         out_tensors = [op.outputs[0] for op in ops]
-        results = self._sess.run(out_tensors, feed_dict={in_tensor: imgs})
-        return list(map(np.squeeze, results))
+        return self._sess.run(out_tensors, feed_dict={in_tensor: imgs})
 
     def _parse_prediction(self, probs):
         idx = np.argmax(probs)
-        label = self.labels_map[idx]
+        label = self.class_labels[idx]
         confidence = probs[idx]
-        return self._package_attr(label, confidence)
+
+        attrs = etad.AttributeContainer()
+        keep = (confidence > self.config.confidence_thresh)
+        if keep:
+            attrs.add(self._package_attr(label, confidence))
+
+        return attrs, keep
+
+    def _package_attr(self, label, confidence):
+        return etad.CategoricalAttribute(
+            self.config.attr_name, label, confidence=confidence)
```

Comment:
This whole function is obsolete

Reply:
what does obsolete mean? The fact that it's only 1 line long? It allows the caller to not wrangle `self.config.attr_name` themselves

Reply:
I suppose...
But it's just moving a single line to another function, and it's only called in one place.
It could just as easily be:
```
packaged_attr = etad.CategoricalAttribute(
    self.config.attr_name, label, confidence=confidence)
attrs.add(packaged_attr)
```

Reply:
Alright its gone